includes/gateway/class-omise-payment-paynow.php

```diff
@@ -70,16 +83,19 @@ public function display_qrcode( $order, $context = 'view' ) {
 		$charge_id = $this->get_charge_id_from_order();
 		$charge    = OmiseCharge::retrieve( $charge_id );
 		$qrcode    = $charge['source']['scannable_code']['image']['download_uri'];
-		?>
-		<div class="omise omise-paynow-details" <?php echo 'email' === $context ? 'style="margin-bottom: 4em; text-align:center;"' : ''; ?>>
-			<div class="omise omise-paynow-logo"></div>
-			<p>
-				<?php echo __( 'Scan the QR code to pay', 'omise' ); ?>
-			</p>
-			<div class="omise omise-paynow-qrcode">
-				<img src="<?php echo $qrcode; ?>" alt="Omise QR code ID: <?php echo $charge['source']['scannable_code']['image']['id']; ?>">
+
+		if ( 'view' === $context ) : ?>
+			<div class="omise omise-paynow-details" <?php echo 'email' === $context ? 'style="margin-bottom: 4em; text-align:center;"' : ''; ?>>
+				<div class="omise omise-paynow-logo"></div>
+				<p>
+					<?php echo __( 'Scan the QR code to pay', 'omise' ); ?>
+				</p>
+				<div class="omise omise-paynow-qrcode">
+					<img src="<?php echo $qrcode; ?>" alt="Omise QR code ID: <?php echo $charge['source']['scannable_code']['image']['id']; ?>">
+				</div>
 			</div>
-		</div>
-		<?php
+		<?php elseif ( 'email' === $context ) : ?>
```

Comment:
This should be excluded from email content if the charge with failed status.

Reply:
From the implementation, buyers will not get into this part of code (thank-you page) if the charge status is failed.
Did you get the result otherwise? (maybe I missed some cases...)